integration-cli/docker_cli_pull_test.go

```diff
@@ -757,3 +758,34 @@ func (s *DockerRegistriesSuite) TestPullFromBlockedRegistry(c *check.C) {
 		c.Fatalf("expected %s and %s to have the same size (%s != %s)", bb2Img.name, bbImg.name, bb2Img.size, bbImg.size)
 	}
 }
+
+func (s *DockerRegistriesSuite) TestPullNeedsAuth(c *check.C) {
+	c.Assert(s.d.StartWithBusybox("--add-registry="+s.regWithAuth.url), check.IsNil)
+
+	repo := fmt.Sprintf("%s/runcom/busybox", s.regWithAuth.url)
+	repoUnqualified := "runcom/busybox"
+
+	out, err := s.d.Cmd("tag", "busybox", repo)
+	c.Assert(err, check.IsNil, check.Commentf(out))
+
+	// this means it needs auth...
+	resp, err := http.Get(fmt.Sprintf("http://%s/v2/", s.regWithAuth.url))
+	c.Assert(err, check.IsNil)
+	c.Assert(resp.StatusCode, check.Equals, http.StatusUnauthorized)
+
+	// login with the registry...
+	out, err = s.d.Cmd("login", "-u", "testuser", "-p", "testpassword", "-e", "test@test.org", s.regWithAuth.url)
```

Comment:
I'd make `testuser` and `testpassword` attributes of the `regWithAuth` registry.